ml-algorithms/src/main/java/org/opensearch/ml/engine/MLEngine.java

```diff
@@ -36,12 +51,58 @@ public class MLEngine {
     private final Path mlCachePath;
     private final Path mlModelsCachePath;
 
-    private final Encryptor encryptor;
+    private Encryptor encryptor;
 
-    public MLEngine(Path opensearchDataFolder, Encryptor encryptor) {
+    private final Gson gson;
+
+    public MLEngine(Path opensearchDataFolder) {
+        gson = new GsonBuilder().setPrettyPrinting().create();
         mlCachePath = opensearchDataFolder.resolve("ml_cache");
         mlModelsCachePath = mlCachePath.resolve("models_cache");
-        this.encryptor = encryptor;
+        initMasterKey();
+    }
+
+
+    private String generateMasterKey() {
```

Comment:
Will this be same on all the nodes in the cluster?
If not a password encrypted in one node can't be decrypted in another.